src/libnotify/notification.cr

```diff
@@ -0,0 +1,74 @@
+require "./c"
+
+class Libnotify::Notification
+  property summary : String?
+  property body : String?
+  property icon_path : String?
+  property app_name : String
+  property timeout : Int32
+  property category : String?
+  property append : Bool
+  property transient : Bool
+  property urgency : Libnotify::C::Notifyurgency
+
+  def initialize(@summary = nil, @body = nil, @icon_path = nil,
+                 @timeout = -1, @category = nil, @urgency = Libnotify::C::Notifyurgency::NotifyUrgencyNormal,
+                 @append = false, @transient = true,
+                 @app_name = "default")
+    init_app!
+    @notify = C.notification_new @summary.to_s, @body.to_s, @icon_path.to_s
+    update
+  end
+
+  def initialize(@summary = nil, @body = nil, @icon_path = nil,
+                 @timeout = -1, @category = nil, @urgency = Libnotify::C::Notifyurgency::NotifyUrgencyNormal,
+                 @append = false, @transient = true,
+                 @app_name = "default", &block)
+    init_app!
+    @notify = C.notification_new @summary.to_s, @body.to_s, @icon_path.to_s
+    yield self
+    update
+  end
+
+  def show(gerror = nil)
+    C.notification_show @notify, gerror
+    self
+  end
+
+  def update
+    C.notification_update @notify, @summary.to_s, @body.to_s, @icon_path.to_s
+    C.notification_set_timeout @notify, @timeout
+    C.notification_set_urgency @notify, @urgency
+    C.notification_set_category @notify, @category.to_s
+    C.notification_set_app_name @notify, @app_name
+    self
+  end
+
+  def to_unsafe
+    @notify
+  end
+
+  # set the timeout at `sec`, in seconds
+  def timeout=(sec : Float::Primitive)
```

Comment:
Similar to Ruby's libnotify I'd like to have a distinction between `Int32` (`Int64` maybe?), `Float` and `String`.
In Crystal, we can implement with method-overloading pretty elegantly so we don't have to this `case/when` dance like in Ruby: https://github.com/splattael/libnotify/blob/5d908930a8cd48a2e46c39235f5daf9d1c67bfff/lib/libnotify/api.rb#L84,L99

WDYT?